tools/dispatch_data.py

```diff
@@ -129,6 +124,12 @@ def main():
         default="info",
         help="To enable log level for debugging purposes. Available options: (Critical, Error, Warning, Info, Debug, Notset)",
     )
+    parser.add_argument(
```

Comment:
why do we need this? `num_parts` is supposed be able to inferred from partition assignments? what if the specified one does not match with partition assignment?

Reply:
With these changes to the input metadata.json file, There is no way to infer 'num_parts' from the input metadata.json schema. 

I think we can get this information from the partitions_meta.json file, in the partitions_dir. 
Code is changed accordingly.